batchflow/models/tf/unet.py

```diff
@@ -87,3 +88,140 @@ def build_config(self, names=None):
+                Parameters for post-processing blocks:
+
+                filters : None, int, list of ints or list of lists of ints
+                    same as encoder/blocks/filters
+
+            upsample : dict
+                Parameters for upsampling (see :func:`~.layers.upsample`).
+
+                filters : int, list of ints or list of lists of ints
+                    same as encoder/blocks/filters
+
+            dense : dict
+                Parameters for dense block on skip connections including parameters for conv_block
+
+                upsample : dict
+                    upsample parameters for block
+
+    for more parameters see (see :class:`~.EncoderDecoder`)
+    """
+    @classmethod
+    def default_config(cls):
+        config = super().default_config()
+        return config
+
+    def build_config(self, names=None):
+        config = super().build_config(names)
+
+        num_stages = config.get('body/encoder/num_stages')
+
+        if config.get('body/decoder/dense/upsample') is None:
+            config['body/decoder/dense/upsample'] = dict(layout='tna')
+
+        if config.get('body/decoder/dense/filters') is None:
+            config['body/decoder/dense/filters'] = [32 * 2**i for i in range(num_stages)]
+
+        if config.get('body/decoder/dense/layout') is None:
+            config['body/decoder/dense/layout'] = 'cna'
+
+        if config.get('body/decoder/dense/kernel') is None:
+            config['body/decoder/dense/kernel'] = 3
+
+        if config.get('body/decoder/dense/combine_op') is None:
+            config['body/decoder/dense/combine_op'] = 'concat'
+
+        return config
+
+    @classmethod
+    def decoder(cls, inputs, name='decoder', **kwargs):
+        decoder_filters = kwargs['upsample']['filters']
+
+        num_stages = kwargs.pop('num_stages') or len(inputs)-2
+
+        combine_op = kwargs['dense']['combine_op']
+        combine_args = {'op': combine_op if isinstance(combine_op, str) else '',
+                        'data_format': kwargs.get('data_format'),
+                        **(combine_op if isinstance(combine_op, dict) else {})}
+
+        semantic_outputs = []
+        n_outputs = kwargs['n_outputs']
+        if isinstance(n_outputs, int):
+            n_outputs = list(range(-n_outputs, 0))
+
+        for i in range(1, num_stages):
+            _inputs = inputs[:i] + [inputs[i]] * 2
+
+            _kwargs = kwargs
+            _kwargs['upsample'] = _kwargs['dense']['upsample']
+            _kwargs['upsample']['filters'] = decoder_filters[-i:]
+            _kwargs['blocks']['filters'] = decoder_filters[-i:]
+
+            outputs = super().decoder(_inputs, name=name,
+                                      return_all=True, **{**_kwargs, 'num_stages': i})
+
+            for j, x in enumerate(inputs[:i]):
+                x = combine([x, outputs[::-1][j]], **combine_args)
+
+                dense_args = {**unpack_args(kwargs['dense'], j, num_stages), **kwargs}
```

Comment:
```suggestion
                dense_args = {**kwargs, **unpack_args(kwargs['dense'], j, num_stages)}
```
Current version seems to be working only because `conv_block` applies essentially the same `unpack_args` under the hood. You can even remove this `unpacking`, but explicit is better that implicit, right?